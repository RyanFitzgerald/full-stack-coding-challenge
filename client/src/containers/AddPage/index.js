import React from 'react';
import Helmet from 'react-helmet';
import { toast } from 'react-toastify';
import { postRequest } from '../../utils/api';

// Import components
import SubHeader from '../../components/SubHeader';
import H1 from '../../components/H1';
import H2 from '../../components/H2';
import P from '../../components/P';
import Upload from '../../components/Upload';
import Button from '../../components/Button';
import Container from './Container';

class AddPage extends React.Component {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
    this.addAddresses = this.addAddresses.bind(this);
  }

  handleSubmit(e) {
    // Stop form submission
    e.preventDefault();

    let data = new FormData();
    data.append('file', this.file.files[0]);

    // Call api
    this.addAddresses(data);
  }

  addAddresses(file) {
    postRequest(`/api/v1/addresses`, file).then(data => {
      console.log(data);
      if (typeof data === 'undefined' || data.status === 500 || data.error) {
        toast.error("An error has occured, please try again.");
        return
      }
      
    });
  }

  render() {
    return (
      <div>
        <Helmet>
          <title>Add Address - Full-stack Coding Challenge - Ambyint</title>
        </Helmet>
        <SubHeader>
          <H1>Add Address</H1>
          <P>Use the form below to add additional addresses from a csv file</P>
        </SubHeader>
        <Container>
          <H2>Upload Address CSV File</H2>
          <form onSubmit={this.handleSubmit} encType="multipart/form-data">
            <Upload name="file" type="file" innerRef={input => this.file = input}/>
            <Button type="submit">Upload Addresses</Button>
          </form>
        </Container>
      </div>
    );
  }
}

export default AddPage;