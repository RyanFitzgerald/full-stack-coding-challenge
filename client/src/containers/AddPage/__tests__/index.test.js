import React from 'react';

import AddPage from '../index';

describe('<AddPage />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(
      <AddPage/>
    ).toJSON();
    
    expect(renderedComponent).toMatchSnapshot();
  });
});