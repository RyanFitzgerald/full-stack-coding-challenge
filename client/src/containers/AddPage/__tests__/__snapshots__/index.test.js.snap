// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPage /> should match the snapshot 1`] = `
.c0 {
  background: #f5f5f5;
  background-image: url(bg.png);
  border-bottom: 1px solid #e6e6e6;
  padding: 100px 0;
  text-align: center;
}

.c1 {
  margin: 0 auto;
  max-width: 960px;
  padding: 15px;
}

.c1:after {
  content: '';
  display: table;
  clear: both;
}

.c2 {
  color: #373958;
  font-size: 2.5em;
  line-height: 1.25;
  margin: 0;
}

.c5 {
  color: #373958;
  font-size: 2em;
  line-height: 1.25;
  margin: 15px 0;
  text-align: center;
}

.c3 {
  margin: 0;
}

.c6 {
  border: 1px solid #e6e6e6;
  display: block;
  margin: 0 auto;
  padding: 5px;
}

.c7 {
  background: #3498db;
  border: 0;
  border-radius: 4px;
  color: #fff;
  cursor: pointer;
  display: block;
  margin: 15px auto 0;
  padding: 15px 50px;
  -webkit-transition: 0.25s ease all;
  transition: 0.25s ease all;
}

.c7:hover {
  background: #236c9c;
}

.c4 {
  padding: 50px 15px;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <h1
        className="c2"
      >
        Add Address
      </h1>
      <p
        className="c3"
      >
        Use the form below to add additional addresses from a csv file
      </p>
    </div>
  </div>
  <div
    className="c4 c1"
  >
    <h2
      className="c5"
    >
      Upload Address CSV File
    </h2>
    <form
      encType="multipart/form-data"
      onSubmit={[Function]}
    >
      <input
        accept=".csv"
        className="c6"
        name="file"
        type="file"
      />
      <button
        className="c7"
        type="submit"
      >
        Upload Addresses
      </button>
    </form>
  </div>
</div>
`;
